### PR TITLE
Allowing getting and setting the host boot flash device via control-plane-agent/MGS

### PR DESCRIPTION
**Pull request by jgallagher**
Also adds a mock `gimlet-hf` server for use on gimletlets that don't have a qspi for the real `gimlet-hf` to use.

### CODE REVIEW

File: drv/mock-gimlet-hf-server/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    /// Selects between QSPI flash chips 1 and 2 (if present)
+    dev_state: HfDevSelect,
+}
+
+impl idl::InOrderHostFlashImpl for ServerImpl {
+    fn read_id(
+        &mut self,
+        _: &RecvMessage,
+    ) -> Result<[u8; 20], RequestError<HfError>> {
+        Ok(*b"mockmockmockmockmock")
```
Comment (mkeeter) [inline drv/mock-gimlet-hf-server/src/main.rs:47]: incredible code 💯

File: task/control-plane-agent/src/update/host_flash.rs [hunk 2]
```diff
@@ -29,6 +29,42 @@ impl HostFlashUpdate {
...
+            .map_err(|err| SpError::UpdateFailed(err as u32))?
+        {
+            HfMuxState::SP => (),
+            HfMuxState::HostCPU => return Err(SpError::UpdateSlotBusy),
+        }
+
+        // Swap to the chosen slot.
+        self.task
+            .set_dev(slot)
+            .map_err(|err| SpError::UpdateFailed(err as u32))?;
```
Comment (mkeeter) [inline task/control-plane-agent/src/update/host_flash.rs:64]: It seems like there's a TOCTOU race here.

Luckily, `self.task.set_dev(slot)` _also_ checks that the mux is valid; we can probably eliminate the first check and just its its result directly

```suggestion
        // Swap to the chosen slot, returning an appropriate error if we don't
        // have control over the host flash.
        self.task
            .set_dev(slot)
            .map_err(|err| match err{
                HfError::NotMuxedToSP => SpError::UpdateSlotBusy,
                _ => SpError::UpdateFailed(err as u32)
            })?;
```

(untested)

Comment (jgallagher) [reply task/control-plane-agent/src/update/host_flash.rs]: Nice catch, fixed.

File: app/gimletlet/app.toml [hunk 3]
```diff
@@ -146,7 +146,10 @@ start = true
 task-slots = ["hash_driver", "hf", "i2c_driver", "rng_driver", "sprot", "sys", "update_server", "user_leds"]
 
 [tasks.hf]
+# If you do not have a gimletlet qspi-let adapter but want to test the hf API
+# (e.g., from control-plane-agent / MGS), switch to `drv-mock-gimlet-hf-server`.
 name = "drv-gimlet-hf-server"
+
```
Comment (mkeeter) [inline app/gimletlet/app.toml:152]: Extreme nit: extra newline here

**Review by mkeeter — APPROVED** [review]

**Pull request merged**